src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -87,20 +76,8 @@ public override int SaveChanges(IList<IUpdateEntry> entries)
     /// <returns>Task that when resolved contains the number of documents affected during saving changes.</returns>
     public override async Task<int> SaveChangesAsync(IList<IUpdateEntry> entries, CancellationToken cancellationToken = default)
     {
-        int docsAffected = 0;
-
-        // TODO: Consider splitting the changes up by type and using the Mongo bulk APIs instead
-
-        foreach (var entry in GetAllChangedRootEntries(entries))
-        {
-            // TODO: Exception handling
-            if (await SaveAsync(entry, cancellationToken))
-            {
-                docsAffected++;
-            }
-        }
-
-        return docsAffected;
+        var updates = ConvertUpdateEntriesToMongoUpdates(entries);
+        return (int)await SaveMongoUpdatesAsync(updates, cancellationToken).ConfigureAwait(false);
```

Comment:
All uses of `await` should have `ConfigureAwait(false)`.

Consider adding the analyzer we use in the driver to enforce this.

Reply:
you can also use [`Fody.ConfigureAwait`](https://github.com/Fody/ConfigureAwait) to automate this